torchtitan/optimizer.py

```diff
@@ -79,27 +118,44 @@ def linear_warmup_linear_decay(
     return curr_adjustment
 
 
+class SchedulersContainer:
+    """Util for calling step on multiple learning rate schedulers needed for virtual pipeline stages"""
+
+    def __init__(self, optimizers, lr_lambda):
+        self.schedulers = []
+        for optimizer in optimizers:
+            self.schedulers.append(LambdaLR(optimizer, lr_lambda=lr_lambda))
+
+    def step(self):
+        for schedulers in self.schedulers:
+            schedulers.step()
+
+
+class SchedulersInBackwardContainer:
+    """Util for calling step on multiple learning rate schedulers when optimizers are in backward"""
+
+    def __init__(self, optimizers, lr_lambda):
+        self.schedulers = []
+        for optimizer in optimizers:
+            schedulers = []
```

Comment:
naming is a bit confusing, let's differentiate schedulers names for `model_parts` and for params in a `model_part`

Reply:
Currently use optimizers -> optim_group -> sub_optim
We could synchronize the names in the next PR when we clean up the optimizer wrappers